src/lightning/pytorch/loops/fetchers.py

```diff
@@ -125,27 +129,16 @@ def __next__(self) -> Any:
                 self.done = not self.batches
         elif not self.done:
             # this will run only when no pre-fetching was done.
-            try:
-                self._fetch_next_batch(self.iterator)
-                # consume the batch we just fetched
-                batch = self.batches.pop(0)
-            except StopIteration as ex:
```

Comment:
This `StopIteration` is now managed in `self._fetch_next_batch`

Reply:
lets add it as comment...